templates/testpmd/resource-specs/sriov-vm1-template.yaml

```diff
@@ -0,0 +1,131 @@
+            requests:
+              storage: ${DV_SIZE}
+    running: false
+    template:
+      spec:
+        domain:
+          cpu:
+            cores: ${{NUM_OF_CORES}}
+          devices:
+            disks:
+            - disk:
+                bus: virtio
+              name: ${NAME}
+            - disk:
+                bus: virtio
+              name: cloudinitdisk
+            interfaces:
+            - masquerade: {}
+              name: default
+            - macAddress: ${SECONDARY_MAC}
+              name: sriov-test-network
+              sriov: {}
+            rng: {}
+          memory:
+            guest: ${MEMORY}
+            hugepages:
+              pageSize: ${VM_HUGEPAGES_PAGE_SIZE}
+          resources:
+            requests:
+              memory: ${MEMORY_OVERHEAD}
+        networks:
+        - name: default
+          pod: {}
+        - multus:
+            networkName: ${NAMESPACE}/sriov-test-network
+          name: sriov-test-network
+        nodeSelector:
+          feature.node.kubernetes.io/network-sriov.capable: "true"
+        terminationGracePeriodSeconds: 180
+        volumes:
+        - dataVolume:
+            name: ${NAME}
+          name: ${NAME}
+        - cloudInitNoCloud:
+            networkData: |
+              ethernets:
+                '1':
+                  match:
+                    macaddress: ${SECONDARY_MAC}
+                  set-name: sriov1
+              version: 2
+            userData: |-
+              #cloud-config
+              chpasswd:
+                expire: false
+              password: ${CLOUD_USER_PASSWORD}
+              runcmd:
+              - sudo echo "vm.nr_hugepages = ${OS_HUGEPAGES_PAGE_SIZE}" >> /etc/sysctl.conf && sysctl -p
+              - sudo sh /usr/sbin/change-nic-driver.sh
+          name: cloudinitdisk
+parameters:
+- description: VM name (also used to name the DataVolume, where the VM image is stored).
+  name: NAME
+  value: sriov-vm1
+- description: Image (source container-disk) URL 
+  name: IMAGE_URL
+  required: true
+- description: Number of cores (CPUs)
+  name: NUM_OF_CORES
+  value: "2"
+- description: SR-IOV interface MAC address
+  name: SECONDARY_MAC
+  generate: expression
+  from: '02:[a-fA-F0-9]{2}:[a-fA-F0-9]{2}:[a-fA-F0-9]{2}:[a-fA-F0-9]{2}:[a-fA-F0-9]{2}'
+- description: Target namespace where to create the VM
+  name: NAMESPACE
+  value: sriov-dpdk-ns
+- description: Storage size for the DataVolume (to store the VM image)
+  name: DV_SIZE
+  value: 10Gi
+- description: Randomized password for the cloud-init user cloud-user
+  from: '[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}'
+  generate: expression
+  name: CLOUD_USER_PASSWORD
+- description: Page size of huge pages
+  displayName: Huge page size
+  name: VM_HUGEPAGES_PAGE_SIZE
+  value: 1Gi
+- description: Page size of huge pages (MiB), set at the VM OS level. Must be set in respect to VM_HUGEPAGES_PAGE_SIZE.
+  displayName: Huge page size MiB
+  name: OS_HUGEPAGES_PAGE_SIZE
```

Comment:
nit: I would rephrase the description to something like
_..hugepages size in Megabytes that will be configured during provision of the guest VM.._
And the parameter to `GUEST_OS_HUGEPAGES_SIZE_MB` or `VM_OS_HUGEPAGES_SIZE_MB`

Reply:
Done